optuna/storages/_rdb/alembic/versions/v2.4.0.a.py

```diff
@@ -0,0 +1,80 @@
+"""empty message
+
+Revision ID: v2.4.0.a
+Revises: v1.3.0.a
+Create Date: 2020-11-17 02:16:16.536171
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "v2.4.0.a"
+down_revision = "v1.3.0.a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "study_direction",
+        sa.Column("study_direction_id", sa.Integer(), nullable=False),
+        sa.Column(
+            "direction",
+            sa.Enum("NOT_SET", "MINIMIZE", "MAXIMIZE", name="studydirection"),
+            nullable=False,
+        ),
+        sa.Column("study_id", sa.Integer(), nullable=True),
+        sa.Column("objective_id", sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["study_id"],
+            ["studies.study_id"],
+        ),
+        sa.PrimaryKeyConstraint("study_direction_id"),
+        sa.UniqueConstraint("study_id", "objective_id"),
+    )
+    op.create_table(
+        "trial_intermediate_values",
+        sa.Column("trial_intermediate_values_id", sa.Integer(), nullable=False),
+        sa.Column("trial_id", sa.Integer(), nullable=True),
+        sa.Column("step", sa.Integer(), nullable=True),
+        sa.Column("value", sa.Float(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["trial_id"],
+            ["trials.trial_id"],
+        ),
+        sa.PrimaryKeyConstraint("trial_intermediate_values_id"),
+        sa.UniqueConstraint("trial_id", "step"),
+    )
+    with op.batch_alter_table("studies", schema=None) as batch_op:
+        batch_op.drop_column("direction")
+
+    with op.batch_alter_table("trial_values", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("objective_id", sa.Integer(), nullable=True))
+        batch_op.create_unique_constraint("value_constraint", ["trial_id", "objective_id"])
```

Comment:
I manually added the name of the constraint to avoid `ValueError: Constraint must have a name`.
```python
        #before: automatically generated by alembic
        batch_op.create_unique_constraint(None, ["trial_id", "objective_id"])
        
        #after:
        batch_op.create_unique_constraint("value_constraint", ["trial_id", "objective_id"])
```